types/webpack-dev-middleware/index.d.ts

```diff
@@ -32,6 +33,7 @@ declare namespace WebpackDevMiddleware {
 		serverSideRender?: boolean;
 		logger?: Logger;
 		filename?: string;
+		writeToDisk?: boolean;
```

Comment:
The docs you linked to state that this can also be a function.

> This option also accepts a Function value, which can be used to filter which files are written to disk. 

So this should really be typed as:

```ts
writeToDisk?: boolean | ((filename: string) => boolean);
```